swan-lake/learn-the-platform/ballerina-persist/persist-cli.md

```diff
@@ -0,0 +1,147 @@
+
+```ballerina
+configurable string host = ?;
+configurable int port = ?;
+configurable string user = ?;
+configurable string password = ?;
+configurable string database = ?;
+```
+
+* The script file(script.sql) will contain the scripts to create the tables in the data store. This script file will be generated based on the data store specified in the `Ballerina.toml` file.
+
+Additionally, this command will create/update `Config.toml` file with configurables used to connect the client with the data store. Generated configurables will be based on the data store specified in the `Ballerina.toml` file.
+
+Behaviour of the `generate` command,
+- User should invoke the command within a Ballerina project
+- The user should have initiated `bal persist` feature in the project and update the model definition file.
+- The model definition file should contain the `persist` module import (`import ballerina/persist as _;`)
+- The Model definition file should contain at least one entity
+- If the user invokes the command twice, it will not fail. It will regenerate the files again.
+
+## Generating Migration scripts for the Model Definition Changes[Experimental]
+
+>**Info:** The support for migrations is currently an experimental feature, and its behavior may be subject to change in future releases. Also, the support for migrations is currently limited to MySQL data store.
+
+This command is used to generate the migration scripts for the model definition changes. This command is executed in the project root directory. This command will generate the migration scripts based on the changes in the model definition file. The generated migration scripts will be added to the `migrations` directory inside the `persist` directory.
+
+```bash
+$ bal persist migrate add_employee
+```
+
+The file structure of the project after executing the command will be as follows,
+
+```
+rainier
+   ├── generated
+         └── store
+               ├── persist_client.bal
+               ├── persist_db_config.bal
+               ├── persist_types.bal
+               └── script.sql
+   ├── persist
+         └── model.bal
+         └── migrations
+               └── 20200820120000_add_employee
+                  ├── model.bal
+                  └── script.sql     
+   ├── Ballerina.toml
+   ├── Config.toml
+   └── main.bal
+```
+
+This command will create a new directory for the migration with the timestamp and the label provided by the user. The directory will contain the model definition file and the script file. The script file will contain the SQL script to update the database schema. The model definition file will be the snapshot of the model definition file at the time of executing the command. The next time the user executes the command, it will compare the model definition file with the latest snapshot file and generate the SQL script to update the database schema. So the user should not modify the snapshot file.
```

Comment:
```suggestion
This command will create a new directory for the migration with the timestamp and the label provided by the user. The directory will contain the model definition file and the script file. The script file will contain the SQL script to update the database schema. The model definition file will be the snapshot of the model definition file at the time of executing the command. The next time the user executes the command, it will compare the model definition file with the latest snapshot file and generate the SQL script to update the database schema. So the user should not modify the snapshot file.
```
```suggestion
This command will create a new directory for the migration with the timestamp and the label provided by the user. The directory will contain the model definition file and the script file. The script file will contain the SQL script to update the database schema. The model definition file will be the snapshot of the model definition file at the time of executing the command. The next time you execute the command, it will compare the model definition file with the latest snapshot file and generate the SQL script to update the database schema. Therefore, you should not modify the snapshot file.
```